src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/DataException.cs

```diff
@@ -22,27 +23,22 @@ internal static class ExceptionBuilder
         // The resource Data.txt will ensure proper string text based on the appropriate
         // locale.
 
-        [BidMethod] // this method accepts BID format as an argument, this attribute allows FXCopBid rule to validate calls to it
         static private void TraceException(
-                string trace,
-                [BidArgumentType(typeof(String))] Exception e)
+                string trace, Exception e)
         {
             Debug.Assert(null != e, "TraceException: null Exception");
             if (null != e)
             {
-                Bid.Trace(trace, e.Message);
-                if (Bid.AdvancedOn)
+                SqlClientEventSource._log.Trace(e.Message);
+                try
                 {
-                    try
-                    {
-                        Bid.Trace(", StackTrace='%ls'", Environment.StackTrace);
-                    }
-                    catch (System.Security.SecurityException)
-                    {
-                        // if you don't have permission - you don't get the stack trace
-                    }
+                    SqlClientEventSource._log.Trace($", StackTrace='{Environment.StackTrace}'");
                 }
-                Bid.Trace("\n");
+                catch (System.Security.SecurityException)
+                {
+                    // if you don't have permission - you don't get the stack trace
+                }
+                SqlClientEventSource._log.Trace("\n");
```

Comment:
I don't think this \n will be helpful. EventSource isn't creating a string message like BID was it's going to dispatch each call as a separate data point. I think this can just be removed as a separate call. I suspect that the two events above should be combined if possible so that the exception message and stack trace are part of a single event not two temporally correlated events.